dspace-api/src/main/java/org/dspace/discovery/IndexEventConsumer.java

```diff
@@ -160,11 +166,12 @@ public void end(Context ctx) throws Exception {
         if (objectsToUpdate != null && handlesToDelete != null) {
 
             // update the changed Items not deleted because they were on create list
-            for (DSpaceObject iu : objectsToUpdate) {
+            for (UUID id : objectsToUpdate) {
                 /* we let all types through here and 
                  * allow the search indexer to make 
                  * decisions on indexing and/or removal
                  */
+                Item iu = itemService.find(ctx, id);
```

Comment:
We cannot assume all UUIDs represent Item objects.  If you look at the original code, DSpaceObject is the thing being indexed. This is because Communities and Collections are *also* indexed in Discovery (so that you can search  on their names/descriptions).